### PR TITLE
Purchase 2.0

### PR DESCRIPTION
**Pull request by luminiferous**
1. Purchase Ui 
2.  Purchase connecting to the listing api
3. Purchase database modified.

### CODE REVIEW

**Review by herrickfeng — COMMENTED** [review]
Checked the front end stuff, David or Agid could you check the back end

**Review by david-de-angelis — COMMENTED** [review]
This is ridiculous, there are 800 lines of reformatted code... I have no idea what youve changed, and what you havent.

File: controllers/listingAPI.js [hunk 1]
```diff
@@ -12,144 +12,114 @@ const auth = require('../utils/authUtil');
...
+                    htmlResponse.fail(req, res, "Could not find the listing that you were looking for :(", "Listing not found");
+                } else {
+                    let result = results[0];
+                    result.imgName = attachmentUtil.getImgPath(result.imgName);
+                    baseView.renderWithAddons(req, res, 'pages/listingResult', { result });
+                }
+            },
+            notFound: () => { htmlResponse.fail(req, res, "Failed to get the listing that you were looking for :(", "Listing Fetch Failure"); }
+        });
+    });
```
Comment (Agid-Kaharuba) [inline controllers/listingAPI.js:34]: Unnecessary change, Nothing changed here except formatting.......

File: controllers/listingAPI.js [hunk 2]
```diff
@@ -12,144 +12,114 @@ const auth = require('../utils/authUtil');
...
+router.get('/search=:query', (req, res) => {
+    console.log('Received search query: ' + req.params.query); // Example params usage.
+
+    listingModel.SearchListings(req.params.query,
+        (results) => {
+            for (var i = 0; i < results.length; i++) {
+                results[i].imgName = attachmentUtil.getImgPath(results[i].imgName);
+            }
+            baseView.renderWithAddons(req, res, 'pages/listing', { listings: results });
+        });
```
Comment (Agid-Kaharuba) [inline controllers/listingAPI.js:48]: Unnecessary change, Nothing changed here except formatting.......

File: controllers/listingAPI.js [hunk 3]
```diff
@@ -12,144 +12,114 @@ const auth = require('../utils/authUtil');
...
+    let currentUser = req.cookies.currentUser;
+    // console.log('currentUser is ' + currentUser);
+    baseView.renderWithAddons(req, res, 'pages/confirmPurchase', { currentUser })
 });
 
-router.get('/paymentSummary', auth.authorizeUser, (req, res) =>
-{
-	baseView.renderWithAddons(req, res, 'pages/paymentSummary');
+router.get('/paymentSummary', auth.authorizeUser, (req, res) => {
+    baseView.renderWithAddons(req, res, 'pages/paymentSummary');
```
Comment (Agid-Kaharuba) [inline controllers/listingAPI.js:125]: Unnecessary change, Nothing changed here except formatting.......

File: controllers/listingAPI.js [hunk 4]
```diff
@@ -12,144 +12,114 @@ const auth = require('../utils/authUtil');
...
+            return;
+        }
+
+        if (isAdmin || user.id == listing.id) {
+            listingModel.modifyListing(listing, {
+                success: () => res.send(jsonResponse.success()),
+                fail: (reason) => res.send(jsonResponse.fail(reason))
+            })
+        }
+    })
```
Comment (Agid-Kaharuba) [inline controllers/listingAPI.js:151]: Unnecessary change, Nothing changed here except formatting.......

**Review by Agid-Kaharuba — COMMENTED** [review]
Recommend to reformat the backend code. Some changes were unnecessary, I have marked them. Ignore the formatting in listingResult.ejs as I plan and have discussed to fix this later.
